src/app/services/auth.guard.ts

```diff
@@ -21,26 +21,54 @@ export class AuthGuard implements CanActivate {
     state: RouterStateSnapshot): Observable<boolean> | Promise<boolean> | boolean {
     this.user = JSON.parse(localStorage.getItem('user'));
```

Comment:
create separate method for verifying user and then  call that function in canactivate and canactivatechild